src/seedu/addressbook/commands/Command.java

```diff
@@ -65,4 +65,9 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+    /* to be inherited by child classes, 
+       returns true if it is a mutating class like add/delete  */
+    public boolean isMutating(){
```

Comment:
You can remove `to be inherited by child classes`, thus leaving it to `Returns true if …`

You can make this method an abstract and leave it empty, instead of throwing an exception.